ExampleMod/Content/Tiles/ExampleCrystal.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.ObjectData;
+
+namespace ExampleMod.Content.Tiles
+{
+    internal class ExampleCrystal : ModTile
+    {
+        public override void SetStaticDefaults()
+        {
+            TileObjectData.newTile.Width = 1;
+            TileObjectData.newTile.Height = 1;
+            TileObjectData.newTile.CoordinateWidth = 16;
+            TileObjectData.newTile.CoordinateHeights = new int[] { 16 };
+            TileObjectData.newTile.CoordinatePadding = 2;
+            TileObjectData.newTile.StyleHorizontal = true;
+            TileObjectData.addTile(Type);
+            Main.tileFrameImportant[Type] = true;
+            Main.tileObsidianKill[Type] = true;
+            Main.tileShine2[Type] = true;
+            Main.tileShine[Type] = 500;
+            Main.tileSpelunker[Type] = true;
+        }
+
+        public override ushort GetMapOption(int i, int j) => (ushort)(Main.tile[i, j].TileFrameX / 18);
+
+        public override void KillTile(int i, int j, ref bool fail, ref bool effectOnly, ref bool noItem)
+        {
+            int toDrop = 0;
+            switch (Main.tile[i, j].TileFrameX / 18)
+            {
+                case 0:
+                    toDrop = Mod.Find<ModItem>("ExampleCrystalItem").Type;
+                    DustType = Mod.Find<ModDust>("Sparkle").Type;
```

Comment:
Use `ModContent.ItemType<ExampleCrystalItem>()` and  `ModContent.DustType<Sparkle>()`